deployment_scripts/v2.4/upgrade.sh

```diff
@@ -64,18 +68,60 @@ if [[ -z "${GIT_BRANCH_NAME}" ]]; then
   export GIT_BRANCH_NAME="main"
 fi
 
-echo Migrate data from CosmosDB to Redis...
-# Needed dependancy to parse yaml files
-# wget https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64 -O /usr/bin/yq &&\
-#     chmod +x /usr/bin/yq
+if [[ "${REDIS_API_URL:-}" != "" ]]; then
+  echo Migrate data from CosmosDB to Redis...
+
+  WORKING_DIR=$(mktemp -d -t cosmotech-api-migration-XXXXXXXXXX)
+  pushd "${WORKING_DIR}"
 
-export COSMOS_DB=phoenix-core
-export COSMOS_KEY=$(kubectl -n "${NAMESPACE}" get secret cosmotech-api-v2 -o yaml | yq -r '.data["application-helm.yml"]' | base64 -d | yq '.csm.platform.azure.cosmos.key')
-export COSMOS_URI=$(kubectl -n "${NAMESPACE}" get secret cosmotech-api-v2 -o yaml | yq -r '.data["application-helm.yml"]' | base64 -d | yq '.csm.platform.azure.cosmos.uri')
-export API_SCOPE=.default
+  export COSMOSDB_DATABASE_NAME=phoenix-core
+  export NAMESPACE="phoenix"
+  export REDIS_INSTANCE="cosmotechredis"
+
+cat <<EOF > cosmosdb_migration_pod.yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: migration-pod
+spec:
+  containers:
+  - name: migration-pod
+    image: ghcr.io/cosmo-tech/platform-migrate-redis:latest
+    env:
+    - name: COSMOSDB_DATABASE_NAME
+      value: ${COSMOSDB_DATABASE_NAME}
+    - name: COSMOSDB_URL
+      value: $(kubectl -n ${NAMESPACE} get secret cosmotech-api-v2 -o yaml | yq -r '.data["application-helm.yml"]' | base64 -d | yq '.csm.platform.azure.cosmos.uri')
+    - name: COSMOSDB_KEY
+      value: $(kubectl -n ${NAMESPACE} get secret cosmotech-api-v2 -o yaml | yq -r '.data["application-helm.yml"]' | base64 -d | yq '.csm.platform.azure.cosmos.key')
+    - name: REDIS_API_URL
+      value: ${REDIS_API_URL}
+    - name: REDIS_SERVER
+      value: "${REDIS_INSTANCE}-master"
+    - name: REDIS_PASSWORD
+      value: $(kubectl -n ${NAMESPACE} get secret ${REDIS_INSTANCE} -o jsonpath="{.data.redis-password}" | base64 --decode)
+  nodeSelector:
+    "cosmotech.com/tier": "db"
+  tolerations:
+  - key: "vendor"
+    operator: "Equal"
+    value: "cosmotech"
+    effect: "NoSchedule"
+  restartPolicy: Never
+  initContainers:
+  - name: download-dependencies
+    image: alpine:latest
+    command: ["sh", "-c", "wget https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64", "-o", "/usr/bin/yq", "&&", "chmod", "+x", "/usr/bin/yq"]
+    securityContext:
```

Comment:
Missing config?

Reply:
We should delete this as there is no more needs for it in current conf